src/Models/BaseElement.php

```diff
@@ -528,6 +536,85 @@ public function getContentForSearchIndex(): string
         return $content;
     }
 
+    /**
+     * Provides content for CMS search
+     */
+    public function getContentForCmsSearch(): string
+    {
+        $textualDatabaseFields = $this->getTextualDatabaseFields();
+        $excludedFields = $this->getFieldsExcludedFromCmsSearch();
+        $contents = [];
+        foreach ($this->config()->get('db') as $fieldName => $fieldType) {
+            if (!in_array($fieldType, $textualDatabaseFields) || in_array($fieldName, $excludedFields)) {
+                continue;
+            }
+            $contents[] = $this->$fieldName;
+        }
+        // Use |#| to delimit different fields rather than space so that you don't
+        // accidentally join results of two columns that are next to each other in a table
+        $delimiter = '|#|';
+        $content = implode($delimiter, $contents);
+
+        // Strips tags and be sure there's a space between words.
+        $content = trim(strip_tags(str_replace('<', ' <', $content)));
+
+        // Allow projects to update content of third-party elements.
+        $content .= $delimiter;
+        $this->extend('updateContentForCmsSearch', $content);
```

Comment:
```suggestion
        $content = trim(strip_tags(str_replace('<', ' <', $content))) . $delimiter;

        // Allow projects to update content of third-party elements.
        $this->extend('updateContentForCmsSearch', $content);
```
This just looks a little tidier

Reply:
I think that's less clear.

The idea of adding the $delimiter on its own line is to make it clearer that it's for the benefit of the extension hook

Reply:
That wasn't at all clear to me - the comment will need to reflect that if that's the intention.

Reply:
Also projects can still munch the content together from several fields without the delimiter, should we pass it in as a second param and add a comment (for folks who search where the hook is used) to mention it's recommended to delimit separate pieces of content with the provided delimiter?

Reply:
That sounds like a sensible idea - it'll make it much less likely for people to forget (or not even know about) delimiting content.

Reply:
Actually given that probably makes sense to pass in $contents array rather than $contents string to the extension hook and then delimit so that projects don't need to worry about it